SimTracker/SiPhase2Digitizer/plugins/DigitizerUtility.h

```diff
@@ -14,61 +14,30 @@
 namespace DigitizerUtility {
   class Amplitude {
   public:
-    Amplitude() : _amp(0.0)
-    {}
-    Amplitude(float amp, float frac) :
-      _amp(amp), _frac(1, frac), _hitInfo() 
-    {
-      // in case of digi from noisypixels
-      // the MC information are removed
-      if (_frac[0] < -0.5)
-	_frac.pop_back();
+    Amplitude() : _amp(0.0) {}
+    Amplitude(float amp) : _amp(amp) {}
+    Amplitude( float amp, const PSimHit* hitp, size_t hitIndex, unsigned int tofBin, float frac) :
+      _amp(amp) {
+      if (frac > -0.5) _simInfoList.push_back({frac,new SimHitInfoForLinks(hitp,hitIndex,tofBin)});
     }
 
-    Amplitude( float amp, const PSimHit* hitp, size_t hitIndex, unsigned int tofBin, float frac) :
-    _amp(amp), _frac(1, frac), _hitInfo(new SimHitInfoForLinks(hitp, hitIndex, tofBin)) {
-      // in case of digi from noisypixels
-      // the MC information are removed
-      if (_frac[0] < -0.5) {
-	_frac.pop_back();
-	_hitInfo->trackIds().pop_back();
-      }
+    Amplitude( float amp, float frac) :
```

Comment:
Hi @suchandradutta - you could simply combine all three of these constructors via some default values to the arguments. That would make this cleaner